dom/base/NodeInfo.cpp

```diff
@@ -152,6 +152,10 @@ void NodeInfo::GetPrefix(nsAString& aPrefix) const {
 
 void NodeInfo::GetNamespaceURI(nsAString& aNameSpaceURI) const {
   if (mInner.mNamespaceID > 0) {
+    if (!nsNameSpaceManager::GetInstance()) {
+      SetDOMStringToNull(aNameSpaceURI);
+      return;
+    }
```

Comment:
No, it's perfectly fine to keep it here. I was just wondering because I recently had some segfaults related to the XPATH thing and forgot to open an issue and didn't have the time to finish debugging them. This should solve my issue too :)